lib/http-server.js

```diff
@@ -64,6 +65,20 @@ function HttpServer(options) {
     res.emit('next');
   });
 
+  if (options.username || options.password) {
+    before.push(function (req, res) {
+      var credentials = auth(req);
+
+      if (credentials && credentials.name === options.username && credentials.pass === options.password) {
```

Comment:
It might be worth using something like [secure-compare](https://www.npmjs.com/package/secure-compare) rather than a simple string comparison? (Secure compare is designed to avoid timing attacks.)

To my mind, I wouldn't be too stress about security (because if I was concerned about security, I probably wouldn't be using basic auth...) but it feels like a good idea to "do it right" (and demonstrate better security practices), and it's quite simple to use a simple secure comparison rather than a `===`.

Feel free to ignore me :-) (I'm just some random on the internet :-p ) Just something that came to mind when I saw this :-)